code/modules/organs/subtypes/augment.dm

```diff
@@ -0,0 +1,401 @@
+			if(action.button)
+				action.button.UpdateIcon()
+
+/obj/item/organ/internal/augment/attack_self(var/mob/user)
+	. = ..()
+
+	if(.)
+
+		if(owner.last_special > world.time)
+			to_chat(owner, SPAN_DANGER("\The [src] is still recharging!"))
+			return FALSE
+
+		if(owner.stat || owner.paralysis || owner.stunned || owner.weakened)
+			to_chat(owner, SPAN_DANGER("You can not use your \the [src] in your current state!"))
+			return FALSE
+
+		if(is_bruised())
+			if(do_bruised_act())
+				return FALSE
+
+		if(is_broken())
+			if(do_broken_act())
+				return FALSE
+
+		owner.last_special = world.time + cooldown
+
+
+/obj/item/organ/internal/augment/proc/do_broken_act()
+	spark(get_turf(owner), 3)
+	return TRUE
+
+/obj/item/organ/internal/augment/proc/do_bruised_act()
+	spark(get_turf(owner), 3)
+	return FALSE
+
+/obj/item/organ/internal/augment/timepiece
+	name = "integrated timepiece"
+	icon_state = "augment-clock"
+	parent_organ = BP_HEAD
+	action_button_name = "Actuvate Integrated Timepiece"
+	activable = TRUE
+	organ_tag = BP_AUG_TIMEPIECE
+	action_button_icon = "augment-clock"
+
+/obj/item/organ/internal/augment/timepiece/attack_self(var/mob/user)
+	. = ..()
+
+	if(.)
+		to_chat(owner, "Hello [user], it is currently: '[worldtime2text()]'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'. Have a lovely day.")
+		if (emergency_shuttle.get_status_panel_eta())
+			to_chat(owner, SPAN_WARNING("Notice: You have one (1) scheduled flight, ETA: [emergency_shuttle.get_status_panel_eta()]."))
+
+/obj/item/organ/internal/augment/pda
+	name = "integrated pda"
+	icon_state = "augment-pda"
+	parent_organ = BP_HEAD
+	action_button_name = "Activate Integrated PDA"
+	activable = TRUE
+	organ_tag = BP_AUG_PDA
+	action_button_icon = "augment-pda"
+	cooldown = 5
+	var/obj/item/device/pda/P
+
+/obj/item/organ/internal/augment/pda/Initialize()
+	. = ..()
+	if(!P)
+		P = new /obj/item/device/pda (src)
+		P.canremove = 0
+
+/obj/item/organ/internal/augment/pda/attack_self(var/mob/user)
+	. = ..()
+	if(.)
+		if(P)
+			if(!P.owner)
+				var/obj/item/card/id/idcard = owner.get_active_hand()
+				if(istype(idcard))
+					P.owner = idcard.registered_name
+					P.ownjob = idcard.assignment
+					P.ownrank = idcard.rank
+					P.name = "Integrated PDA-[P.owner] ([P.ownjob])"
+					to_chat(owner, SPAN_NOTICE("Card scanned."))
+					P.try_sort_pda_list()
+				else
+					to_chat(owner, SPAN_NOTICE("No ID data loaded. Please hold your ID to be scanned."))
+					return
+
+			P.attack_self(user)
+	return
+
+/obj/item/organ/internal/augment/pda/emp_act(severity)
+	..()
+	if(P)
+		P = null
+		qdel(P)
+		if(owner)
+			if(owner.can_feel_pain())
+				to_chat(owner, SPAN_DANGER("You feel something burn inside your head!"))
+
+
+/obj/item/organ/internal/augment/tool
+	name = "retractable widget"
+	action_button_name = "Deploy Widget"
+	action_button_icon = "augment-tool"
+	cooldown = 250
+	var/augment_type
+	var/obj/item/augment
```

Comment:
Technically you don't need these two variables? You can do it all in one I'm pretty sure.